official/resnet/keras/keras_cifar_test.py

```diff
@@ -0,0 +1,193 @@
+        "-distribution_strategy", "off",
+        "-model_dir", "keras_cifar_1_gpu_no_dist_strat",
+    ]
+    extra_flags = extra_flags + self._extra_flags
+
+    integration.run_synthetic(
+        main=keras_cifar_main.run,
+        tmp_root=self.get_temp_dir(),
+        extra_flags=extra_flags
+    )
+
+  def test_end_to_end_graph_1_gpu_no_dist_strat(self):
+    """Test Keras model in legacy graph mode with 1 GPU, no dist strat."""
+    if context.num_gpus() < 1:
+      self.skipTest(
+          "{} GPUs are not available for this test. {} GPUs are available".
+          format(1, context.num_gpus()))
+
+    extra_flags = [
+        "-num_gpus", "1",
+        "-enable_eager", "false",
+        "-distribution_strategy", "off",
+        "-model_dir", "keras_cifar_graph_1_gpu_no_dist_strat",
+    ]
+    extra_flags = extra_flags + self._extra_flags
+
+    integration.run_synthetic(
+        main=keras_cifar_main.run,
+        tmp_root=self.get_temp_dir(),
+        extra_flags=extra_flags
+    )
+
+  def test_end_to_end_1_gpu(self):
+    """Test Keras model with 1 GPU."""
+    if context.num_gpus() < 1:
+      self.skipTest(
+          "{} GPUs are not available for this test. {} GPUs are available".
+          format(1, context.num_gpus()))
+
+    extra_flags = [
+        "-num_gpus", "1",
+        "-enable_eager", "true",
+        "-distribution_strategy", "default",
+        "-model_dir", "keras_cifar_1_gpu",
+    ]
+    extra_flags = extra_flags + self._extra_flags
+
+    integration.run_synthetic(
+        main=keras_cifar_main.run,
+        tmp_root=self.get_temp_dir(),
+        extra_flags=extra_flags
+    )
+
+  def test_end_to_end_graph_1_gpu(self):
+    """Test Keras model in legacy graph mode with 1 GPU."""
+    if context.num_gpus() < 1:
+      self.skipTest(
+          "{} GPUs are not available for this test. {} GPUs are available".
+          format(1, context.num_gpus()))
+
+    extra_flags = [
+        "-num_gpus", "1",
+        "-enable_eager", "false",
+        "-distribution_strategy", "default",
+        "-model_dir", "keras_cifar_graph_1_gpu",
+    ]
+    extra_flags = extra_flags + self._extra_flags
+
+    integration.run_synthetic(
+        main=keras_cifar_main.run,
+        tmp_root=self.get_temp_dir(),
+        extra_flags=extra_flags
+    )
+
+  def test_end_to_end_2_gpu(self):
+    """Test Keras model with 2 GPUs."""
+    if context.num_gpus() < 2:
+      self.skipTest(
+          "{} GPUs are not available for this test. {} GPUs are available".
+          format(2, context.num_gpus()))
+
+    extra_flags = [
+        "-num_gpus", "2",
+        "-enable_eager", "true",
+        "-distribution_strategy", "default",
+        "-model_dir", "keras_cifar_2_gpu",
+    ]
+    extra_flags = extra_flags + self._extra_flags
+
+    integration.run_synthetic(
+        main=keras_cifar_main.run,
+        tmp_root=self.get_temp_dir(),
+        extra_flags=extra_flags
+    )
+
+  def test_end_to_end_graph_2_gpu(self):
```

Comment:
will these tests run on guitar automatically? or do we need to do something? lets make sure they do once submitted.

Reply:
If you are taking about externally on GitHub, they are: https://sponge.corp.google.com/target?id=b0c0dda4-6ac2-4c02-839e-501edc50aef8&target=tensorflow_models%2Fubuntu%2Fpy2%2Fpresubmit&searchFor=
(search for keras_cifar_test). 
Although, it looks like they are all skipped. I think it is because all the tests are GPU tests, they require at least 1 GPU. I changed the 1-GPU tests to CPU, and will kick off another run.

If you are talking about internally on google3, I think Igor has done the work to include the tests in official/ to TAP.

Reply:
I think the CPU tests will run fine. My question is specifically about GPU tests. can you confirm how we can get them to run with 1 GPU both internally and externally? And how about 2 GPUs?